### DISCUSSION TITLE
have a "can't create directory '/var/lib/grafana-plugins': Permission denied" when updating in docker

### DISCUSSION THREAD

**Discussion by Ultra3sndk** [teslamate-org/teslamate#3576] [Q&A]
Hello folks,

everything was running fine until I noticed 2 updates on my Synology docker TESLAMATE app, the TESLAMATE and the GRAFANA.
I started by GRAFANA and it failed, I have the "can't create directory '/var/lib/grafana-plugins': Permission denied" while the TESLAMATE update was fine.

I checked on google, and I cannot say I found useful informations.
Did you encounter this issue and if yes how did you solved it  ?

THANK YOU !

PS : on the Synology disk this dir EXIST. but maybe the error is inside the container, but I cannot enter it as it refuse to launch.

**Comment by IBMRob** [comment]
I'm also hitting the same problems as this, its occurred after a restart of the container and I suspect the version of the Grafana image has been upgraded. Previously it was running a v8.x version of Grafana. 

I can see that `/volume1/docker/teslamate/datagrafana` is mounted into `/var/lib/grafana` and is writable but I don't have anything for '/var/lib/grafana-plugins'`

**Comment by regator99** [comment]
Same issue here after updating container this morning, not sure where to start

**Comment by Ultra3sndk** [comment]
answer to myself : 
As I cannot find out how to solve that the proper way, I backuped the Database (the whole directory), deleted all containers, reinstalled everything (with DB in v14 not v15), copied back the DB dir and I found all the datas back.

**Comment by IBMRob** [comment]
I managed to fix this by updating the `GF_PATHS_PLUGINS` env var to be `/var/lib/grafana/plugins` instead of `/var/lib/grafana-plugins`. This updated it to use one of the folders on the shares synology drive so it should be retained for reboots etc

**Reply by Faltek123** [reply]
It worked for me, but does not display data, any advice ?

**Reply by regator99** [reply]
Changed the path several days ago, can now view grafana, but "no data" on all dashboards

**Reply by regator99** [reply]
the error I am seeing is "Status: 500. Message: db query error: pq: SSL is not enabled on the server" on all my dashboards 

update, after using  docker-compose pull &
  docker-compose up -d

all seems to be working now, will update if things change.

**Comment by stergios1** [comment]
Here is how i solved this problem. Note: I am not saving docker data in /var/lib/docker (this is the default)

1. change the volume path in compose.yaml under grafana variables, as follows:
    volumes:
      - /opt/stacks/teslamate/teslamate-grafana-data/_data:/var/lib/grafana

2. check the permissions in _data directory:
$ ls -l teslamate-grafana-data/_data
total 2428
drwx------ 2 472 root    4096 Mar 29 18:09 csv
-rw-r----- 1 472 root 2465792 Mar 30 14:31 grafana.db
drwx------ 2 472 root    4096 Mar 29 18:09 pdf
drwxrwxrwx 2 472 root    4096 Feb 23 23:29 plugins
drwx------ 2 472 root    4096 Mar 29 18:09 png
drwxr-x--- 3 472 root    4096 Mar 29 18:09 unified-search

If you are using the default volumes directory, the structure should be like this:
$ sudo ls -l /var/lib/docker/volumes/teslamate_teslamate-grafana-data
total 4
drwxrwxrwx 7 472 root 4096 Mar 30 14:19 _data
$ sudo ls -l /var/lib/docker/volumes/teslamate_teslamate-grafana-data/_data
total 2432
drwx------ 2 472 root    4096 Mar 29 18:09 csv
-rw-r----- 1 472 root 2465792 Mar 30 14:19 grafana.db
drwx------ 2 472 root    4096 Mar 29 18:09 pdf
drwxrwxrwx 2 472 root    4096 Feb 23 23:29 plugins
drwx------ 2 472 root    4096 Mar 29 18:09 png
drwxr-x--- 3 472 root    4096 Mar 29 18:09 unified-search
